quill-engine/src/main/scala/io/getquill/sql/norm/ExpandDistinct.scala

```diff
@@ -17,19 +17,32 @@ object ExpandDistinct {
       case Distinct(q) =>
         trace"ExpandDistinct Distinct(inside)" andReturn
           Distinct(apply(q))
+      case DistinctOn(q, a, b) =>
```

Comment:
Wait a minute... I think DistinctOn clauses should not actually be here. Please try to take them out.